links/src/commonMain/kotlin/com/splendo/kaluga/links/state/LinksStateRepo.kt

```diff
@@ -0,0 +1,143 @@
+                            }
+                        }
+                    }
+                    is LinksState.Ready<*> -> {
+                        when (link) {
+                            is Links.Incoming.Result<*> -> {
+                                { state.ready(link.data) }
+                            }
+                            is Links.Failure -> {
+                                { state.error(link.message) }
+                            }
+                            is Links.Outgoing.Link -> {
+                                { state.open(link.url) }
+                            }
+                            Links.Pending -> {
+                                { state.pending() }
+                            }
+                        }
+                    }
+                    is LinksState.Pending -> {
+                        when (link) {
+                            is Links.Incoming.Result<*> -> {
+                                { state.ready(link.data) }
+                            }
+                            is Links.Failure -> {
+                                { state.error(link.message) }
+                            }
+                            is Links.Outgoing.Link -> {
+                                { state.open(link.url) }
+                            }
+                            Links.Pending -> {
+                                state.remain()
+                            }
+                        }
+                    }
+                    is LinksState.Open -> {
+                        when (link) {
+                            is Links.Incoming.Result<*> -> {
+                                { state.ready(link.data) }
+                            }
+
+                            is Links.Failure -> {
+                                { state.error(link.message) }
+                            }
+
+                            is Links.Outgoing.Link -> {
+                                { state.open(link.url) }
+                            }
+                            Links.Pending -> {
+                                { state.pending() }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Convert an incoming url into an object and return it wrapped in [Links.Incoming.Result].
+     * When the given Uri/NSURL is invalid, the state returns [Links.Failure].
+     * @param query query containing the values from the link.
+     * @param serializer The needed object's serializer.
+     * */
+    fun <T> handleIncomingLink(query: String, serializer: KSerializer<T>) {
+        linksManager?.handleIncomingLink(query, serializer)
+    }
+
+    /**
+     * Check if the url is valid and returns [Links.Outgoing.Link] with the given url. State returns [Links.Failure]
+     * when the url is invalid.
+     * @param url Page to be visited.
+     * */
+    fun handleOutgoingLink(url: String) {
```

Comment:
From the code I see it's just validates given string if it's correct URL and that's it.
So it should be called `validateLink` or open in external browser?
Or even accept a base url and map of parameters to build (serialize) valid url?

Reply:
I agree with you that this needs to be renamed into something else since it just validate the url and return Error or Success. Then `kaluga-architecture` will be in charge of opening the browser. Calling it `validateLink` is fine to me.